crates/types/src/tokens.rs

```diff
@@ -43,3 +43,50 @@ impl TryFrom<SqliteRow> for TokenMetadata {
         })
     }
 }
+
+#[derive(Debug, Serialize)]
+pub struct Erc721Token {
+    pub info: Erc721TokenInfo,
+    pub name: String,
+    pub symbol: String,
+    pub uri: String,
+}
+
+#[derive(Debug, Serialize)]
+pub struct Erc721TokenInfo {
+    pub contract: Address,
+    pub owner: Address,
+    pub token_id: U256,
+}
+
+#[derive(Debug, Serialize)]
+pub struct Erc721TokenMetadata {
+    pub name: String,
+    pub symbol: String,
+    pub uri: String,
+}
+
+impl TryFrom<SqliteRow> for Erc721TokenInfo {
+    type Error = ();
+
+    fn try_from(row: SqliteRow) -> Result<Self, Self::Error> {
+        Ok(Self {
+            contract: Address::from_str(row.get("contract")).unwrap(),
+            owner: Address::from_str(row.get("owner")).unwrap(),
+            token_id: U256::from_str(row.get("token_id")).unwrap(),
+        })
+    }
+}
+
+impl TryFrom<SqliteRow> for Erc721Token {
+    type Error = ();
+
+    fn try_from(row: SqliteRow) -> Result<Self, Self::Error> {
+        Ok(Self {
+            name: row.get("name"),
+            symbol: row.get("symbol"),
+            uri: row.get("uri"),
+            info: row.try_into().unwrap(),
```

Comment:
```suggestion
            info: row.try_into()?,
```

we should avoid unwraps, especially in these contexts